docs/src/reference/message-correlation.md

```diff
@@ -1,35 +1,43 @@
 
-When a workflow instance enters a message catch event then it polls the buffer for a proper message. If a proper message exists then it is correlated to the workflow instance. In case multiple messages match the correlation information then the first published message is correlated. The behavior of the buffer is similar to a queue.    
+Messages have a TTL, so boundary events can be triggered in processes that start after the arrival of a message, if the message TTL has not yet expired when the process starts.
 
-The buffering of a message is disabled when its TTL is set to zero. If the message can't be correlated to a workflow instance then it is discarded.
+The buffering of a message is disabled when its TTL is set to zero. In this case, if the message can't be correlated to a running workflow instance immediately when it arrives, it is discarded.
 
 ## Message Uniqueness
 
-A message can contain a unique id to ensure that it is published only once (i.e. idempotent). The id can be any string, for example, a request id, a tracking number or the offset/position in a message queue.
+A message can be optionally guaranteed to be idempotent within the window of its TTL. If a message contains a unique id, this is combined with the name and correlationKey value and compared with other messages that are still in the buffer (those whose TTL has not expired). The id can be any string - for example, a request id, a tracking number or the offset/position in a message queue.
 
-When the message is published then it checks if a message with the same name, correlation key and id exists in the buffer. If yes then the message is rejected and not correlated.
+When an idempotent message is published, the broker checks the buffer if a message with the same name, correlation key and id has been received. If it has, then the message is rejected and not correlated. Messages are removed from the buffer when their TTL has expired. So messages are only idempotent within the window of their TTL.
 
-> Note that the uniqueness check only looks into the buffer. If a message is discarded from the buffer then a message with the same name, correlation key and id can be published afterward.
+> Note that the uniqueness check only looks into the buffer. If a message is discarded from the buffer then a message with the same name, correlation key and id can be published and correlated after a previous one has expired. If you need 100% guaranteed idempotency over a long period of time, set a long TTL on the message. If the volume of such messages is high, it will impact performance, so you should design your TTL around the expected lifetime of processes, and move the idempotency check outside the broker if it is extremely long.
```

Comment:
This is a good addition but I still would not recommend setting a long TTL :sweat_smile: 

> If you need 100% guaranteed idempotency over a long period of time, set a long TTL on the message.